### PR TITLE
First PR for unit tests

### PR DESCRIPTION
**Pull request by salv-orlando**
They're kind of useful, right?

### CODE REVIEW

**Comment by shettyg** [comment]
Noob question: How do I run the unit tests from terminal?

File: ovn_k8s/tests/unit/test_kubernetes.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import json
```
Comment (russellb) [inline ovn_k8s/tests/unit/test_kubernetes.py]: Can you add a license header here?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 pbr>=1.8
 netaddr>=0.7.18
 requests>=2.9.1
-ovs
+ovs; python_version < '3.0'
+ovs==2.6.0.dev3; python_version >= '3.0'
```
Comment (russellb) [inline requirements.txt]: ovs>=2.6.0.dev3 would be good, as I'm about to post 2.6.0 to pypi.

**Comment by russellb** [comment]
It looks like travis-ci is running the py27 and flake8 environments.  Could you update it so it runs the py35 environment, as well?

**Comment by salv-orlando** [comment]
Sure @russellb 
I did not enable 3.5 before as we had no unit test and did not want to waste resources.
(also they fail right now -:) )

@shettyg you can do as Russell says or just run 
```
$ tox
```
for running flake8, python 2.7 and python 3.5

**Comment by salv-orlando** [comment]
@russellb  #40

File: ovn_k8s/tests/unit/test_kubernetes.py (hunk 1 above)
Comment (salv-orlando) [inline ovn_k8s/tests/unit/test_kubernetes.py]: sure!

File: requirements.txt (hunk 2 above)
Comment (salv-orlando) [inline requirements.txt]: I think we can just do ovs>=2.6.0 now

**Comment by salv-orlando** [comment]
PR has been updated to reflect changes in common/kubernetes.py

Also added more tests to ensure coverage for auth and certificate verify.

**Comment by shettyg** [comment]
I don't have anything intelligent to comment as I have never used mock. lgtm.

File: ovn_k8s/common/kubernetes.py [hunk 3]
```diff
@@ -105,7 +105,7 @@ def get_pod_annotations(server, namespace, pod):
         response = requests.get(url, headers=headers, verify=ca_certificate)
     else:
         response = requests.get(url, headers=headers)
-    if not response or response.status_code != 200:
+    if not response:
```
Comment (russellb) [inline ovn_k8s/common/kubernetes.py]: Ideally I'd prefer this type of change to be in its own bug fix commit and not include in the unit tests commit.

**Comment by salv-orlando** [comment]
I moved the changes into common/kubernetes.py into a distinct commit.

**Comment by russellb** [comment]
:+1:

**Pull request merged**